scripts/reboot

```diff
@@ -134,11 +134,19 @@ fi
 if [ -x ${DEVPATH}/${PLATFORM}/${PLAT_REBOOT} ]; then
     VERBOSE=yes debug "Rebooting with platform ${PLATFORM} specific tool ..."
     exec ${DEVPATH}/${PLATFORM}/${PLAT_REBOOT} $@
-else
-    # If no platform-specific reboot tool, just run /sbin/reboot
-    exec /sbin/reboot $@
+
+    # There are a couple reasons execution reaches here:
+    #
+    # 1. The vendor platform reboot returned after scheduled the platform specific reboot.
+    #    This is a vendor platform reboot code bug but it happens.
+    # 2. The vendor platform reboot failed. e.g. due to platform driver didn't load properly.
+    #
+    # As result if the reboot script reaches here. We should make the reboot happen.
+    # Sleep 1 second before calling /sbin/reboot to accommodate situation #1 above.
+    sleep 1
+
+    VERBOSE=yes debug "Reboot failed!" >&2
 fi
 
-# Should never reach here
-VERBOSE=yes debug "Reboot failed!" >&2
-exit 1
+exec /sbin/reboot $@
```

Comment:
Suggest to also echo another message to the user before issuing /sbin/reboot (and possibly also log it), like "Issuing OS-level reboot..."

Reply:
This is a common path when lack of platform reboot tool. Added suggested wording above.

Reply:
Understood, but if the user sees the message "Platform-specific reboot failed!" from above, I think it would also be nice to inform them that the system is still going to reboot. Plus, if this is logged, we can determine if for some reason a platform-specific reboot is not attempted when we believe it should have benn.

Reply:
I initially thought the common successful path should be quiet. But since we logged issuing platform specific reboot anyways. I guess it doesn't hurt to make it symmetric.